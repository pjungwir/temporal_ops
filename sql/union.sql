SELECT  id, UNNEST(range_agg(valid_at)) AS valid_at
FROM  (
  SELECT * FROM a
  UNION
  SELECT * FROM b
) x
GROUP BY id
ORDER BY id, valid_at;
