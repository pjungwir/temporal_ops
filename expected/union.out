SELECT  id, UNNEST(range_agg(valid_at)) AS valid_at
FROM  (
  SELECT * FROM a
  UNION
  SELECT * FROM b
) x
GROUP BY id
ORDER BY id, valid_at;
 id | valid_at  
----+-----------
  1 | [1,30)
  2 | [1,20)
  3 | [5,10)
  4 | [1,20)
  4 | [500,600)
  6 | [1,20)
  7 | [5,20)
  8 | [5,10)
  9 | [1,20)
(9 rows)

